src/checkstyle/checkstyle.xml

```diff
@@ -1,152 +0,0 @@
-<?xml version="1.0"?>
-<!DOCTYPE module PUBLIC
```

Comment:
My strong preference here is to keep the rules and correct the behaviour for `this` as well as unused imports. Also, wrap them into an artifact so that the reported issues around reactor and subproject builds are avoided.